### PR TITLE
[sidechannel]: checkSignature() with votes (Mat 1342)

### PR DESCRIPTION
**Pull request by AyushKaul**
tests for more than 2/3+1 and less than 2/3+1

### CODE REVIEW

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 1]
```diff
@@ -1192,6 +1193,126 @@ contract('StakeManager', async function (accounts) {
     })
   })
 
+  describe('checkSignature() with votes', function () {
```
Comment (imyourm8) [inline test/units/staking/stakeManager/StakeManager.test.js:1196]: just "checkSignature"

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 2]
```diff
@@ -1192,6 +1193,126 @@ contract('StakeManager', async function (accounts) {
...
+      this.validatorsWallets = {}
+      this.totalStaked = new BN(0)
+      this.accumulatedFees = {}
+
+      for (let i = 0; i < this.validatorsCount; i++) {
+        await approveAndStake.call(this, { wallet: wallets[i], stakeAmount: amount })
+
+        const validatorId = i + 1
+        this.validatorsWallets[validatorId] = wallets[i]
+        console.log(this.validatorsWallets[validatorId].getAddressString(), validatorId)
```
Comment (imyourm8) [inline test/units/staking/stakeManager/StakeManager.test.js:1239]: remove console.log please

File: test/units/staking/stakeManager/StakeManager.test.js (hunk 1 above)
Comment (imyourm8) [reply test/units/staking/stakeManager/StakeManager.test.js:1196]: Ah, no, there is already "checkSignatures" describe block. You must move entire test case inside that scope. The structure i would make would be:
```
describe('with votes')
     describe('with 4 validators')
          describe('when 2/3 + 1 vote yes`)
               it('checkpoints')
          describe('when 2/3 vote yes')
               it('reverts')
```

Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js:1196]: okay, i'll change it.

File: test/units/staking/stakeManager/StakeManager.test.js (hunk 2 above)
Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js:1239]: sure

File: test/units/staking/stakeManager/StakeManager.test.js (hunk 1 above)
Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js]: done

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 3]
```diff
@@ -1192,6 +1192,127 @@ contract('StakeManager', async function (accounts) {
     })
   })
 
+  describe.only('checkSignature() with votes', function () {
```
Comment (imyourm8) [inline test/units/staking/stakeManager/StakeManager.test.js:1195]: Almost there, now move this into existing checkSignatures and rename it to "with votes" and dont forget to remove .only()

Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js]: done

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 4]
```diff
@@ -342,6 +343,85 @@ contract('StakeManager', async function (accounts) {
         testRevert()
       })
     })
+
+    describe('checkSignatures with votes', function () {
```
Comment (imyourm8) [inline test/units/staking/stakeManager/StakeManager.test.js:347]: just "with votes" it is already part of checkSignatures

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 5]
```diff
@@ -342,6 +343,85 @@ contract('StakeManager', async function (accounts) {
...
+        const secondFeeToClaim = new BN(web3.utils.toWei('100'))
+  
+        before(function () {
+          this.trees = []
+          this.validatorsCount = 4
+        })
+        before('fresh deploy', doDeploy)
+
+        describe('Alice proposes with more than 2/3+1 votes votes', function () {
+          it('should pass the check', async function () {
```
Comment (0xAshish) [inline test/units/staking/stakeManager/StakeManager.test.js:412]: add 2 more tests for 2/3+1 votes and sigPrefix[0x00, 0x02]

Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js:412]: done

File: test/units/staking/stakeManager/StakeManager.test.js (hunk 4 above)
Comment (AyushKaul) [reply test/units/staking/stakeManager/StakeManager.test.js]: done

File: test/units/staking/stakeManager/StakeManager.test.js [hunk 6]
```diff
@@ -342,6 +343,93 @@ contract('StakeManager', async function (accounts) {
...
+        const AliceValidatorId = 1
+        const totalFee = new BN(web3.utils.toWei('100'))
+        const firstFeeToClaim = new BN(web3.utils.toWei('25'))
+        const secondFeeToClaim = new BN(web3.utils.toWei('100'))
+  
+        before(function () {
+          this.trees = []
+          this.validatorsCount = 4
+        })
+        before('fresh deploy', doDeploy)
```
Comment (imyourm8) [inline test/units/staking/stakeManager/StakeManager.test.js:409]: change it to beforeEach please

**Pull request merged**